ibis/expr/operations/relations.py

```diff
@@ -127,12 +127,13 @@ def dtype(self):
 @public
 class ScalarSubquery(Subquery):
     def __init__(self, rel):
-        from ibis.expr.rewrites import ReductionValue
+        from ibis.expr.operations import Reduction
 
         super().__init__(rel=rel)
-        if not self.value.find(ReductionValue, filter=Value):
+        if not isinstance(self.value, Reduction):
```

Comment:
The wrapped value must be a reduction instance.

Reply:
That is equivalent with `ops.ScalarSubquery(t.agg(avg=t.x.mean()).avg) - 2` and we prefer to have the scalar subquery wrapper as close to the reduction as possible.